src/m365/spo/commands/file/file-sharinglink-clear.ts

```diff
@@ -0,0 +1,170 @@
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+    this.#initOptionSets();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        fileUrl: typeof args.options.fileUrl !== 'undefined',
+        fileId: typeof args.options.fileId !== 'undefined',
+        scope: typeof args.options.scope !== 'undefined',
+        confirm: !!args.options.confirm
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-u, --webUrl <webUrl>'
+      },
+      {
+        option: '--fileUrl [fileUrl]'
+      },
+      {
+        option: '--fileId [fileId]'
+      },
+      {
+        option: '--scope [scope]',
+        autocomplete: this.allowedScopes
+      },
+      {
+        option: '--confirm'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        const isValidSharePointUrl: boolean | string = validation.isValidSharePointUrl(args.options.webUrl);
+        if (isValidSharePointUrl !== true) {
+          return isValidSharePointUrl;
+        }
+
+        if (args.options.fileId && !validation.isValidGuid(args.options.fileId)) {
+          return `${args.options.fileId} is not a valid GUID`;
+        }
+
+        if (args.options.scope && this.allowedScopes.indexOf(args.options.scope) === -1) {
+          return `'${args.options.scope}' is not a valid scope. Allowed values are: ${this.allowedScopes.join(',')}`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push({ options: ['fileUrl', 'fileId'] });
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    const clearSharingLinks: () => Promise<void> = async (): Promise<void> => {
+      try {
+        if (this.verbose) {
+          logger.logToStderr(`Clearing sharing links for file ${args.options.fileUrl || args.options.fileId}${args.options.scope ? ` with scope ${args.options.scope}` : ''}`);
+        }
+
+        const fileDetails = await spo.getFileDetails(args.options.webUrl, args.options.fileId, args.options.fileUrl);
+        const sharingLinks = await this.getFileSharingLinks(args.options.webUrl, logger, args.options.fileId, args.options.fileUrl, args.options.scope);
+        const batchRequests = sharingLinks.map((sharingLink, index) => {
+          return {
+            id: index,
+            method: 'DELETE',
+            url: `/sites/${fileDetails.SiteId}/drives/${fileDetails.VroomDriveID}/items/${fileDetails.VroomItemID}/permissions/${sharingLink.id}`
+          };
+        });
+        for (let i = 0; i < batchRequests.length; i += 20) {
+          const batchRequestChunk = batchRequests.slice(i, i + 20);
+          const requestOptions: CliRequestOptions = {
+            url: 'https://graph.microsoft.com/v1.0/$batch',
```

Comment:
I like the idea that we are batching all requests. However sometimes I'm getting weird behavior. When I have multiple links, some links break instead of being removed.

![afbeelding](https://user-images.githubusercontent.com/11723921/215904330-ec03c3b6-bf9e-41ad-979f-ba220484b186.png)

If I remove the batching and process all requests one by one, it does seem to remove all links normally. So I'm afraid we should process all requests one by one.